### DISCUSSION TITLE
How to limit body length only given http path?

### DISCUSSION THREAD

**Discussion by doowonee** [tokio-rs/axum#2243] [Q&A] [answered]
I know axum can limit body easily but I want to enlarge the body limit only specific path.

Could you give me any advice for me?

**Comment by davidpdrsn** [comment]
Just add the middleware for that path specifically. See https://docs.rs/axum/latest/axum/middleware/index.html#applying-middleware for relevant links.

**Reply by doowonee** [reply]
Thanks for your comment.

In my case only start with `/upload` path handler limit 50mb and rest of it like  `/user` hanlders limit default as 2mb is it possible?

It is using DefaultBodyLimit duplicated so I thought it's wrong.

**Reply by davidpdrsn** [reply, marked as answer]
> In my case only start with /upload path handler limit 50mb and rest of it like  /user hanlders limit default as 2mb is it possible?

Yes that's possible. Just only add the middleware where you need it

```rust
Router::new()
    .route(
        "/upload",
        post(upload).layer(DefaultBodyLimit::max(52_428_800)),
    )
    .route("/other-route", get(foobar));
```

**Reply by doowonee** [reply]
You saved my life! thank you!!!
